src/otx/algo/instance_segmentation/maskdino.py

```diff
@@ -0,0 +1,517 @@
+            param_groups,
+            optimizer.defaults["lr"],
+        )
+
+        schedulers = self.scheduler_callable(optimizer_with_grad_clip)
+
+        def ensure_list(item: Any) -> list:  # noqa: ANN401
+            return item if isinstance(item, list) else [item]
+
+        lr_scheduler_configs = []
+        for scheduler in ensure_list(schedulers):
+            lr_scheduler_config = {"scheduler": scheduler}
+            if hasattr(scheduler, "interval"):
+                lr_scheduler_config["interval"] = scheduler.interval
+            if hasattr(scheduler, "monitor"):
+                lr_scheduler_config["monitor"] = scheduler.monitor
+            lr_scheduler_configs.append(lr_scheduler_config)
+
+        return [optimizer_with_grad_clip], lr_scheduler_configs
+
+    def _get_optim_params(self, model: nn.Module) -> list[dict[str, Any]]:
+        """Get optimizer parameters."""
+        _optimizer = self.optimizer_callable(self.parameters())
+
+        # Configurable from MaskDINO recipe
+        base_lr = _optimizer.defaults["lr"]
+        weight_decay = _optimizer.defaults["weight_decay"]
+        defaults = {
+            "lr": base_lr,
+            "weight_decay": weight_decay,
+        }
+
+        # Static optimizer params
+        weight_decay_norm = 0.0
+        weight_decay_embed = 0.0
+        backbone_multiplier = 0.1
+
+        norm_module_types = (
+            torch.nn.BatchNorm1d,
+            torch.nn.BatchNorm2d,
+            torch.nn.BatchNorm3d,
+            torch.nn.SyncBatchNorm,
+            torch.nn.GroupNorm,
+            torch.nn.InstanceNorm1d,
+            torch.nn.InstanceNorm2d,
+            torch.nn.InstanceNorm3d,
+            torch.nn.LayerNorm,
+            torch.nn.LocalResponseNorm,
+        )
+
+        params = []
+        uniques = set()
+        for module_name, module in model.named_modules():
+            for module_param_name, value in module.named_parameters(recurse=False):
+                if not value.requires_grad:
+                    continue
+                # Avoid duplicating parameters
+                if value in uniques:
+                    continue
+                uniques.add(value)
+
+                hyperparams = copy.copy(defaults)
+                if "backbone" in module_name:
+                    hyperparams["lr"] = hyperparams["lr"] * backbone_multiplier
+                if "relative_position_bias_table" in module_param_name or "absolute_pos_embed" in module_param_name:
+                    hyperparams["weight_decay"] = 0.0
+                if isinstance(module, norm_module_types):
+                    hyperparams["weight_decay"] = weight_decay_norm
+                if isinstance(module, torch.nn.Embedding):
+                    hyperparams["weight_decay"] = weight_decay_embed
+                params.append({"params": [value], **hyperparams})
+        return params
+
+    @staticmethod
+    def _add_grad_clipping(
```

Comment:
Isn't it possible to use `gradient_clip_val` of torchlightning?

Reply:
I think it's possible, but I plan to carry out further experiments and fine-tuning in the next phase.

Reply:
If so, how about using `gradient_clip_val` and tuning that value?